spec/mongo/index/view_spec.rb

```diff
@@ -196,7 +196,70 @@
         expect(view.get('random_name')).to_not be_nil
       end
     end
-  end
+
+    context 'when providing an invalid partial index filter' do
```

Comment:
In general, the Ruby driver differs from the Python driver in that we don't rigorously check input from users, nor do we test different invalid input scenarios, unless it's our code that does the validation.
If we do test invalid user input, we usually only check one type and verify that an OperationFailure is properly raised. So here, I would only keep [this test](https://github.com/mongodb/mongo-ruby-driver/pull/697/files#diff-5fc639ad0e4b26a0c396bab4704c3fceR206) and take out the other invalid cases.